src/Microsoft.DotNet.Build.CloudTestTasks/AzureHelper.cs

```diff
@@ -182,6 +186,95 @@ public static List<string> GetHeaderValues(HttpRequestHeaders headers, string he
             return list;
         }
 
+        private static bool IsWithinRetryRange(HttpStatusCode statusCode)
+        {
+            // Retry on http 'server error' codes (5xx) as well as redirect
+
+            var rawStatus = (int)statusCode;
+            if (rawStatus == 302)
+                return true;
+            else if (rawStatus >= 500 && rawStatus <= 599)
```

Comment:
I don't see anything in the Helix code that retries on these failures (RETRY_RANGE_CLIENT_ERROR).  I can add it however I'd like to see some precedent before blindly adding.  @ChadNedzlek what's your take.